src/containers/pages/FocusInvestigation/active/index.tsx

```diff
@@ -175,19 +196,122 @@ class ActiveFocusInvestigation extends React.Component<
     }
 
-    if (
-      caseTriggeredPlans &&
-      caseTriggeredPlans.length === 0 &&
-      routinePlans &&
-      routinePlans.length === 0 &&
-      searchedTitle === null
-    ) {
+    if (this.state.loading) {
       return <Loading />;
     }
-    const routineReactivePlans: Dictionary[] = [];
+
     const pageTitle = jurisdictionName
       ? format(FI_IN_JURISDICTION, jurisdictionName)
       : CURRENT_FOCUS_INVESTIGATION;
+
+    const locationColumns: Array<Column<Dictionary>> = getLocationColumns(locationHierarchy);
+    const commonColumns: Array<Column<Dictionary>> = [
+      {
+        Cell: (cell: Cell<Dictionary>) => {
+          const original = cell.row.original;
+          return (
+            <div>
+              {original.focusArea.trim() && (
+                <Link to={`${FI_SINGLE_MAP_URL}/${original.id}`}>{cell.value}</Link>
+              )}
+              &nbsp;
+              <NewRecordBadge recordDate={original.plan_date} />
+            </div>
+          );
+        },
+        Header: NAME,
+        accessor: 'plan_title',
+        minWidth: 180,
+      },
+      {
+        Header: FI_STATUS,
+        accessor: (d: Dictionary) => planStatusDisplay[d.plan_status] || d.plan_status,
+        id: 'plan_status',
+        minWidth: 80,
+      },
+      ...locationColumns,
+      {
+        Cell: (cell: Cell<Dictionary>) => {
+          const original = cell.row.original;
+          return (
+            <div>
+              {original.focusArea.trim() && cell.value}
+              &nbsp;&nbsp;
+              {original.focusArea.trim() && (
+                <Link to={`${FI_SINGLE_URL}/${original.jurisdiction_id}`}>
+                  <FontAwesomeIcon icon={['fas', 'external-link-square-alt']} />
+                </Link>
+              )}
+            </div>
+          );
+        },
+        Header: FOCUS_AREA_HEADER,
+        accessor: 'focusArea',
+        minWidth: 180,
+      },
+      {
+        Header: STATUS_HEADER,
+        accessor: 'status',
+        maxWidth: 60,
+      },
+    ];
+
+    const caseTriggeredColumns = [
+      ...commonColumns,
+      {
+        Cell: ({ value }: Cell<Dictionary>) => {
+          return <div>{value}</div>;
+        },
+        Header: CASE_NOTIF_DATE_HEADER,
+        accessor: 'caseNotificationDate',
+        minWidth: 90,
+      },
+      {
+        Header: CASE_CLASSIFICATION_HEADER,
+        accessor: 'caseClassification',
+      },
+    ];
+    const RoutineColumns = [
+      ...commonColumns,
+      {
+        Cell: ({ value }: Cell<Dictionary>) => {
+          return <div>{value}</div>;
+        },
+        Header: START_DATE,
+        accessor: 'plan_effective_period_start',
+        minWidth: 80,
+      },
+      {
+        Header: END_DATE,
+        accessor: 'plan_effective_period_end',
+      },
+    ];
+
+    const createTableProps = <D extends object>(
+      columns: Array<Column<D>>,
+      data: Plan[] | null,
+      queryParam: string
+    ) => {
```

Comment:
@p-netm would it make sense to move this out of the render function so we don't re-declare it every render? Also, docstrings pls 🙏

Reply:
cool